llvm/lib/CodeGen/SelectionDAG/LegalizeVectorTypes.cpp

```diff
@@ -4448,6 +4470,8 @@ void DAGTypeLegalizer::WidenVectorResult(SDNode *N, unsigned ResNo) {
     if (!TLI.isOperationLegalOrCustomOrPromote(N->getOpcode(), WideVecVT) &&
         TLI.isOperationExpand(N->getOpcode(), VT.getScalarType())) {
       Res = DAG.UnrollVectorOp(N, WideVecVT.getVectorNumElements());
+      if (N->getNumValues() == 2)
```

Comment:
For operations that return > 2 values (which I appreciate you might not be able to test), this doesn't do the right thing now. If you change this condition to `> 1`, then the code in `ReplaceOtherWidenResult` should handle the case the operation has more results. This currently fails the assert, but I think can be implemented with a loop for the other values.

Reply:
I'm not sure this should be implemented now. `DAG.UnrollVectorOp()` also asserts that the number of results is 1 or 2. I could just add an assert after the `if` that the number of results is 1 and leave further extension to when someone needs it?

Reply:
I guess not -- I've generalized this, but yeah, there are no nodes that test this path right now.